File: src/main/java/play/pluv/playlist/exception/PlayListExceptionType.java
package play.pluv.playlist.exception;

import static org.springframework.http.HttpStatus.NOT_FOUND;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import play.pluv.base.BaseExceptionType;

@RequiredArgsConstructor
@Getter
public enum PlayListExceptionType implements BaseExceptionType {

  PLAYLIST_PROVIDER_NOT_FOUND(NOT_FOUND, "지원하지 않는 스트리밍 서비스입니다");

  private final HttpStatus httpStatus;
  private final String message;
}

File: src/main/kotlin/play/pluv/playlist/exception/PlayListExceptionType.kt
package play.pluv.playlist.exception

import org.springframework.http.HttpStatus
import play.pluv.base.BaseExceptionType

enum class PlayListExceptionType(
    override val httpStatus: HttpStatus,
    override val message: String
) : BaseExceptionType {

    PLAYLIST_PROVIDER_NOT_FOUND(HttpStatus.NOT_FOUND, "지원하지 않는 스트리밍 서비스입니다");

}
